### PR TITLE
Johnstill/rw 240 fuzzy finder

### PR DESCRIPTION
**Pull request by johnstill**
This is likely _not_ the final PR, but is a good chunk of functionality to merge.  The final version of fuzzy as-you-type search will likely include some debouncing, and possibly cancellation of old requests.  I need https://github.com/all-of-us/workbench/pull/286 to put the finish on this task.

### CODE REVIEW

File: api/src/main/resources/workbench.yaml [hunk 1]
```diff
@@ -647,12 +647,12 @@ paths:
       description: Quick search for criteria that matches on code and name.
       operationId: "getCriteriaTreeQuickSearch"
       parameters:
-        - in: path
+        - in: query
```
Comment (freemabd1) [inline api/src/main/resources/workbench.yaml]: I think that we want this to be path? So we can call the api like so: http://all-of-us-workbench-test.appspot.com/api/v1/cohortbuilder/quicksearch/ICD9/cho ... Also, will we have a min characters before we calls this api?

**Comment by freemabd1** [comment]
Also, I'm assuming that you need to merge changes from https://github.com/all-of-us/workbench/pull/286 and pull into the api call for quick search

File: api/src/main/resources/workbench.yaml (hunk 1 above)
Comment (johnstill) [reply api/src/main/resources/workbench.yaml]: The API call is not made by the frontend until there are three or more chars in the search string (that check is [here](https://github.com/all-of-us/workbench/pull/289/files#diff-caa6588446cbd142d58bd51a9b342050R67))

I think these should be query parameters.  But if they should be path parameters the path needs to be altered to reflect that.

Comment (freemabd1) [reply api/src/main/resources/workbench.yaml]: the api is implemented to use path.

**Comment by johnstill** [comment]
I'm merging this - there may be some optimizations to be made (as noted in the code) but those will be their own PR.

**Pull request merged**